src/local-introspection.js

```diff
@@ -1,67 +1,72 @@
 const debug = require('debug')('token-introspection');
+const JwksClient = require('jwks-rsa');
 const jwk2pem = require('pem-jwk').jwk2pem;
 const jwt = require('jsonwebtoken');
 const promisify = require('util.promisify');
 
 const jwtVerify = promisify(jwt.verify);
 
-function findCandidateKeys(jwtHeader, keys) {
-  function alg2keyType(alg) {
-    if (alg.startsWith('HS')) {
-      return 'oct';
-    } else if (alg.startsWith('RS') || alg.startsWith('PS')) {
-      return 'RSA';
-    } else if (alg.startsWith('ES')) {
-      return 'EC';
-    }
-    return null;
+module.exports = (options) => {
+  let jwksFetchKey = null;
+  if (options.jwks) {
+    debug('Configured JWKS with static keys');
+    jwksFetchKey = (key) => {
+      const k = options.jwks.keys.find(obj => obj.kid === key);
+      if (k) {
+        return Promise.resolve(Object.assign({ rsaPublicKey: jwk2pem(k) }, k));
+      }
+      return Promise.reject('Unable to find key');
```

Comment:
[Wrap reason string in `Error`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject#Description).